### ISSUE TITLE
bug: Sort function not work very well with hungarian letters

### ISSUE THREAD

**Issue by og900aero** [eza-community/eza#1006]
$ eza --version
v0.18.14 [+git]

command that I using:
eza -lahg --color=always --group-directories-first

Debian 12 stable.

Sort of accented characters does not yet work properly.

![screenshot_1716806555](https://github.com/eza-community/eza/assets/49487376/208edb4a-c8c9-443a-ba2c-c60f05403c33)

The correct sequence is always the following regardless of what the second character contains

A....
Á....
...
E
É
...
I
Í
...
O
Ó
Ö
Ő
...
U
Ú
Ü
Ű
...

---

**Comment by daviessm** [comment]
It looks like we're using the [`natord` crate](https://docs.rs/natord/latest/natord/) to do the sorting, which doesn't handle most of Unicode properly. However, it's not an easy problem to solve: https://sts10.github.io/2023/01/29/sorting-words-alphabetically-rust.html

This might be a good first issue for someone new to the project to work on.

**Comment by og900aero** [comment]
> It looks like we're using the [`natord` crate](https://docs.rs/natord/latest/natord/) to [do the sorting](https://github.com/eza-community/eza/blob/22dbeec229e8221dade471b411395b38299f9fc7/src/fs/filter.rs#L240), which doesn't handle most of Unicode properly. However, it's not an easy problem to solve: https://sts10.github.io/2023/01/29/sorting-words-alphabetically-rust.html
> 
> This might be a good first issue for someone new to the project to work on.

sxyazi success under rust in his file manager: [https://github.com/sxyazi/yazi](https://github.com/sxyazi/yazi)
Maybe he can help.

**Comment by LeoniePhiline** [comment]
Is there still interest to have work done on this issue?

By a happy little accident, I came across [my own WIP PR from 2024](https://github.com/eza-community/eza/pull/1048), which I had inadvertently abandoned. (Apologies!)

I had a look at https://github.com/cafkafk/natord-plus-plus/pull/1, which is stalled on review, but it seems to me that a `locale` option [is required to get this right](https://sts10.github.io/2023/01/29/sorting-words-alphabetically-rust.html#resigning-myself-to-the-fact-that-we-need-to-add-a-locale-option-to-tidy).

I had started this approach by [parsing an `icu_locid::Locale` from `${LANG}`](https://github.com/eza-community/eza/pull/1048/changes/9d8d435025956bc91b34b8d35f634c0dace2b2a8#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR84-R102), and I would move on by [using a locale-aware `icu::collator::Collator` while sorting naturally](https://sts10.github.io/2023/01/29/sorting-words-alphabetically-rust.html#november-2023-update-a-single-crate-dependency).

Given the current state of https://github.com/cafkafk/natord-plus-plus/pull/1, I do not know if I should continue work on #1048 – but I would advise not to go forward with a locale-unaware natural sorting algorithm as proposed in https://github.com/cafkafk/natord-plus-plus/pull/1, and neither one which uses [manual normalization tables](https://github.com/cafkafk/natord-plus-plus/pull/1/changes#diff-0544fc952eefa8d170bbd200b75827f5bb58e39b132e7c871c1bc8a9d5e09ce5R190-R229).

[Normalization](https://docs.rs/icu/latest/icu/normalizer/index.html) and [collation](https://docs.rs/icu/latest/icu/collator/index.html) are exactly what `icu` is for.

**Comment by og900aero** [comment]
> Is there still interest to have work done on this issue?
> 
Yes. The problem still exists, and it would be good to find a solution.

**Comment by MuntasirSZN** [comment]
Agreed, `icu` should be used, I took a look at `yazi`'s implementation and opened a pr.

**Comment by cafkafk** [comment]
Duplicate of #1448 — root cause is the same: eza doesn't honor LC_COLLATE, which is why accented/non-ASCII letters (Hungarian in this case) don't sort correctly. Tracking under #1448.
